src/blocked.c

```diff
@@ -767,59 +539,12 @@ void blockForKeys(client *c, int btype, robj **keys, int numkeys, long count, ms
             l = dictGetVal(de);
         }
         listAddNodeTail(l,c);
-        bki->listnode = listLast(l);
+        dictAdd(c->bstate.keys,keys[j],listLast(l));
     }
+    c->flags |= CLIENT_PENDING_COMMAND;
     blockClient(c,btype);
 }
 
-/* Unblock a client that's waiting in a blocking operation such as BLPOP.
- * You should never call this function directly, but unblockClient() instead. */
-void unblockClientWaitingData(client *c) {
```

Comment:
can you please reinstate these moved functions to the same location they were (perhaps at the price of a forward declaration)? it's hard to review